plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,718 @@
+    return numbers[1], numbers[0]
+
+
+def find_mount_target(module, target, results):
+    """Execute df command to access the information of mount points.
+
+    Parameters
+    ----------
+        module : object
+            Ansible object to execute commands.
+        target : str
+            ZFS to check.
+
+    Returns
+    -------
+        mount_point : str
+            The folder where the zfs is mount
+    """
+    rc, stdout, stderr = module.run_command("df")
+    found = False
+    if rc == 0:
+        stdout_lines = stdout.split("\n")
+        for line in stdout_lines:
+            if len(line) > 2:
+                columns = line.split()
+                if target in columns[1]:
+                    mount_point = columns[0]
+                    found = True
+                    break
+        if found is False:
+            module.fail_json(msg="No mount points were found in the following output: {0}".format(stdout), **results)
+    return mount_point
+
+
+def convert_size(size, space_type):
+    """Function to convert size from kb to the space_type.
+
+    Parameters
+    ----------
+        size : int
+            Size to grow or shrink the zfs
+        space_type : str
+            Type of space to be use
+
+    Returns
+    -------
+        size : int
+            Size on kilobytes
+    """
+    if space_type == "m":
+        size /= 1024
+    if space_type == "g":
+        size /= 1048576
+    if space_type == "cyl":
+        size /= 830
+    if space_type == "trk":
+        size /= 55
+    return size
+
+
+def proper_size_str(size, space_type):
+    """Function to convert size to a proper response for output.
+
+    Parameters
+    ----------
+        size : int
+            Size to grow or shrink the zfs
+        space_type : str
+            Type of space to be use
+
+    Returns
+    -------
+        size : int or float
+            Size on specific space_type notation 2 if is 2.0 or 1.5
+    """
+    if space_type == "k" or space_type == "trk" or space_type == "cyl":
+        return int(size)
+    else:
+        split_size = str(size).split(".")
+        if split_size[1].startswith("0"):
+            return int(size)
+        else:
+            return float("{:.1f}".format(size))
+
+
+def create_trace_dataset(name, member=False):
+    """Function to create data sets for traceback if is not created.
+
+    Parameters
+    ----------
+        name : str
+            Full size of the dataset
+        member : bool
+            If the dataset include a member to create
+
+    Returns
+    -------
+        rc : bool
+            Indicates if datasets were made.
+    """
+    if member:
+        dataset_name = data_set.extract_dsname(name)
+        data_set.DataSet.ensure_present(name=dataset_name, replace=False, type="PDS", record_length=400)
+        rc = data_set.DataSet.ensure_member_present(name)
+    else:
+        rc = data_set.DataSet.ensure_present(name=name, replace=False, type="SEQ", record_length=400)
```

Comment:
Curious, if SEQ cause a false negative, why are we promoting that , why not use a PDS (without a member) instead of SEQ, PDS don't have to have members.